cmd/common-main.go

```diff
@@ -962,12 +962,14 @@ func handleKMSConfig() {
 		if err != nil {
 			logger.Fatal(err, "Unable to initialize a connection to KES as specified by the shell environment")
 		}
-		// We check that the default key ID exists or try to create it otherwise.
-		// This implicitly checks that we can communicate to KES. We don't treat
-		// a policy error as failure condition since MinIO may not have the permission
+		// Try to generate a data encryption key. Only try to create key if this fails.
+		// This implicitly checks that we can communicate to KES.
+		// We don't treat a policy error as failure condition since MinIO may not have the permission
 		// to create keys - just to generate/decrypt data encryption keys.
-		if err = KMS.CreateKey(context.Background(), env.Get(kms.EnvKESKeyName, "")); err != nil && !errors.Is(err, kes.ErrKeyExists) && !errors.Is(err, kes.ErrNotAllowed) {
-			logger.Fatal(err, "Unable to initialize a connection to KES as specified by the shell environment")
+		if _, err = KMS.GenerateKey(GlobalContext, env.Get(kms.EnvKESKeyName, ""), kms.Context{}); err != nil && errors.Is(err, kes.ErrKeyNotFound) {
```

Comment:
Let's say if KES server (not KMS) itself is offline, it doesn't seem we call logger.Fatal() in that case, I think it was failing before this change, can you please confirm ?

Reply:
@vadmeste I just tested. If KES is offline then `KMS.CreateKey` fails

Reply:
e.g. `ERROR Unable to initialize a connection to KES as specified by the shell environment: kes: connection error: Post "https://10.214.226.181:9073/v1/key/create/minio-key": dial tcp 10.214.226.181:9073: connect: connection refused`

Reply:
Please update the comment with relevant permissions needed on the kes/kms side such as 

```
allow: 
 - /v1/<name>/*
```